ndcube/ndcube.py

```diff
@@ -444,42 +444,56 @@ def quantity(self):
         return u.Quantity(self.data, self.unit, copy=_NUMPY_COPY_IF_NEEDED)
 
     def _generate_world_coords(self, pixel_corners, wcs, needed_axes=None, *, units):
-        # Create meshgrid of all pixel coordinates.
-        # If user wants pixel_corners, set pixel values to pixel corners.
+        # This will generate only the coordinates that are needed if there is no correlation within the WCS
+        # This bypasses the entire rest of the function below which works out the full set of coordinates
+        # This only works for WCS that have the same number of world and pixel dimensions
+        if not pixel_corners and needed_axes is not None and not isinstance(wcs, ExtraCoords) and np.sum(wcs.axis_correlation_matrix[needed_axes]) == 1 and len(self.data.shape) == wcs.world_n_dim:
```

Comment:
For future developers (and reviewer), could you add comments stating why each of these conditions are required?  This my clarify some questions I have, such as:
* why does this only work when `not pixel_corners`?
* Does this only work when there is only one `needed_axis`?
  * If not, what is the significance of the magic `0` index below, e.g. `wcs.axis_correlation_matrix[needed_axes][0]]`

Reply:
> For future developers (and reviewer), could you add comments stating why each of these conditions are required? This may clarify some questions I have, such as:

Fair

> * why does this only work when `not pixel_corners`?

Too stupid to work out how to fix it.

> * Does this only work when there is only one `needed_axis`?

Too stupid to work out how to fix it.
   
>   * If not, what is the significance of the magic `0` index below, e.g. `wcs.axis_correlation_matrix[needed_axes][0]]`

I think `wcs.axis_correlation_matrix[needed_axes]` returns a list that has to be escaped hence the [0]. But I should check.

Reply:
> > * why does this only work when `not pixel_corners`?
> 
> Too stupid to work out how to fix it.

See suggested change below for initial guess at how to solve this.  Can you write a test to check out if this in fact does fix it?

> > * Does this only work when there is only one `needed_axis`?
> 
> Too stupid to work out how to fix it.

Would this work if you made the unwanted dimensions for indices have shape `(1) * n` where `n` is the number of dimensions.  And the indices in the wanted dimensions can have the `np.arange` output.  Since the dimensions are independent, this should result in broadcastable inputs to `wcs.pixel_to_world_values`, and so you should be able to get coords out for multiple independent axes, without blowing up the memory.  At least, I think this is true.

> > * If not, what is the significance of the magic `0` index below, e.g. `wcs.axis_correlation_matrix[needed_axes][0]]`
> 
> I think `wcs.axis_correlation_matrix[needed_axes]` returns a list that has to be escaped hence the [0]. But I should check.

Is this list length 1 because you enforced this condition above: `np.sum(wcs.axis_correlation_matrix[needed_axes]) == 1`?

Reply:
The reason I added that was to avoid the hack I wrote breaking here:

```
  File "/home/nabil/Git/ndcube/.tox/py/lib/python3.13/site-packages/ndcube/ndcube.py", line 451, in _generate_world_coords
    lims = (-0.5, self.data.shape[::-1][needed_axes[0]] + 1) if pixel_corners else (0, self.data.shape[::-1][needed_axes[0]])
```
due to
```
ValueError: operands could not be broadcast together with shapes (1,6) (5,)
```

Reply:
In the one case in the test suite, no this is not enough.

`(0, self.data.shape[::-1][needed_axes[0]])` errors with `*** IndexError: tuple index out of range` as the "self.data.shape" is length 3 but the needed_axis is "3". There is no direct translation between the needed world coordinate and the corresponding pixel coordinate in this block.